hazelcast/src/main/java/com/hazelcast/map/impl/proxy/NearCachedMapProxyImpl.java

```diff
@@ -443,23 +444,22 @@ public void onInvalidate(Invalidation invalidation) {
         }
 
         @Override
-        public void handle(BatchNearCacheInvalidation batchNearCacheInvalidation) {
-            List<SingleNearCacheInvalidation> invalidations = batchNearCacheInvalidation.getInvalidations();
+        public void handle(BatchNearCacheInvalidation batch) {
+            List<Invalidation> invalidations = batch.getInvalidations();
 
-            for (SingleNearCacheInvalidation invalidation : invalidations) {
-                handle(invalidation);
+            for (Invalidation invalidation : invalidations) {
+                invalidation.consumedBy(this);
```

Comment:
Could we perhaps remove this level of indirection? Seems like all `Invalidation` classes just call `invalidationHandler.handle(this)` and we just return the control back to the handler.